internal/controller/factory/statefulset.go

```diff
@@ -161,6 +274,44 @@ func generateEtcdArgs(cluster *etcdaenixiov1alpha1.EtcdCluster) []string {
 		args = append(args, fmt.Sprintf("%s=%s", flag, value))
 	}
 
+	peerTlsSettings := []string{"--peer-auto-tls"}
+
+	if cluster.Spec.Security != nil && cluster.Spec.Security.Peer != nil {
+		peerTlsSettings = []string{
+			"--peer-trusted-ca-file=/etc/etcd/pki/peer/ca/ca.crt",
+			"--peer-cert-file=/etc/etcd/pki/peer/cert/tls.crt",
+			"--peer-key-file=/etc/etcd/pki/peer/cert/tls.key",
+			"--peer-client-cert-auth",
+		}
+	}
+
+	serverProtocol := "http"
+
+	clientTlsSettings := []string{"--auto-tls"}
+
+	if cluster.Spec.Security != nil && cluster.Spec.Security.ClientServer != nil {
+		clientTlsSettings = []string{
+			"--trusted-ca-file=/etc/etcd/pki/server/ca/ca.crt",
+			"--cert-file=/etc/etcd/pki/server/cert/tls.crt",
+			"--key-file=/etc/etcd/pki/server/cert/tls.key",
+			"--client-cert-auth=false",
```

Comment:
If `--trusted-ca-file` is not empty, it makes no sense to set `--client-cert-auth` to false, as `--trusted-ca-file` is NOT the CA, that signed `--cert-file`, but the CA that should sign client certificates when mTLS is enabled.

Reply:
Wow. This is bug. I missed it. It should not have =false. Thank you.